### ISSUE TITLE
Feature request: protect member access of optional fields

### ISSUE THREAD

**Issue by dallonf** [palantir/tslint#685]
It would be amazingly helpful if there was a rule that could detect situations like this:

```ts
interface Person {
  address?: {
    line1: string
  }
}

let person = getPerson();
console.log(person.address.line1); // `person.address` might be null or undefined here
```

but would allow:

```ts
if (person.address) {
  console.log(person.address.line1);
}
// or
console.log(person.address && person.address.line1);
```

and, of course, would allow the access of members of required fieds, no questions asked.

**Comment by adidahiya** [comment]
related to, if not the same as, #326

**Comment by dallonf** [comment]
Ah, I didn't see that one, it is very similar - although that rule seems to be focused on preventing an implicit cast of optional to required values; my suggestion focuses on preventing inline TypeErrors in chained access. They could very well be the same rule, though; they have the exact same mechanism of detecting unprotected use of an optional value. Should I move this suggestion to a comment on #326?

**Comment by JKillian** [comment]
Feel free to comment in the other thread, I think both these cases should be encompassed into one rule, perhaps something like `safe-nullable-access` or `safe-optional-access`. Probably best to keep it all in one issue.

As a side note, [TypeScript has a proposal for an operator to accomplish this](https://github.com/Microsoft/TypeScript/issues/16)!
